src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -468,11 +472,17 @@ private void handleCloudWatchAttemptStatus(CloudWatchAttempt cloudWatchAttempt)
         });
         currentProcessingLogFilePerComponent.forEach(componentCurrentProcessingLogFile::put);
 
-        componentCurrentProcessingLogFile.forEach((componentName, currentProcessingFileInformation) -> {
-            Topics componentTopics = getRuntimeConfig()
-                    .lookupTopics(PERSISTED_COMPONENT_CURRENT_PROCESSING_FILE_INFORMATION, componentName);
-            componentTopics.replaceAndWait(currentProcessingFileInformation.convertToMapOfObjects());
+        context.runOnPublishQueueAndWait(() -> {
+            componentCurrentProcessingLogFile.forEach((componentName, currentProcessingFileInformation) -> {
+                Topics componentTopics =
+                        getRuntimeConfig().lookupTopics(PERSISTED_COMPONENT_CURRENT_PROCESSING_FILE_INFORMATION,
+                                componentName);
+                componentTopics.updateFromMap(currentProcessingFileInformation.convertToMapOfObjects(),
+                        new UpdateBehaviorTree(UpdateBehaviorTree.UpdateBehavior.REPLACE, System.currentTimeMillis()));
+            });
         });
+        context.waitForPublishQueueToClear();
```

Comment:
nice - I was wondering if the publish was somehow responsible for the issue

Reply:
Since I could not reproduce the issue, I'm not entirely sure that it is, but I just want to remove it as a possible source.

Reply:
yeah - makes sense to batch up the updates though